### PR TITLE
*: follower support to handle `GetRegion` and other region api

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #7431
should be merged after #7489 #7611
### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Integration test
- Manual test (add detailed scripts or steps below)
image

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
PD allows followers to handle gRPC API about region info such as `GetRegion` and `ScanRegions`. It can increase the maximum carrying capability of the PD routing service.
```

### CODE REVIEW

**Comment by CabinfeverB** [comment]
cc @niubell

File: server/grpc_service.go [hunk 1]
```diff
@@ -1960,10 +2026,25 @@ func (s *GrpcServer) GetOperator(ctx context.Context, request *pdpb.GetOperatorR
 }
 
 // validateRequest checks if Server is leader and clusterID is matched.
-// TODO: Call it in gRPC interceptor.
 func (s *GrpcServer) validateRequest(header *pdpb.RequestHeader) error {
-	if s.IsClosed() || !s.member.IsLeader() {
-		return ErrNotLeader
+	return s.validateRoleInRequest(context.TODO(), header, nil)
```
Comment (rleungx) [inline server/grpc_service.go:2030]: Why not just name it as validateRequest?

File: server/grpc_service.go [hunk 2]
```diff
@@ -1297,22 +1303,39 @@ func (s *GrpcServer) GetRegion(ctx context.Context, request *pdpb.GetRegionReque
...
-		return &pdpb.GetRegionResponse{Header: s.notBootstrappedHeader()}, nil
-	}
-	region := rc.GetRegionByKey(request.GetRegionKey())
-	if region == nil {
-		return &pdpb.GetRegionResponse{Header: s.header()}, nil
+	var rc *cluster.RaftCluster
+	var region *core.RegionInfo
+	if *followerHandle {
+		rc = s.cluster
+		if !s.cluster.GetRegionSyncer().IsRunning() {
```
Comment (rleungx) [inline server/grpc_service.go:1316]:
```suggestion
		if !rc.GetRegionSyncer().IsRunning() {
```

File: server/grpc_service.go (hunk 1 above)
Comment (CabinfeverB) [reply server/grpc_service.go:2030]: Because some code needs original `validateRequest` which is only one param and is easy to use, so I keep it.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**